### PR TITLE
1269 pass blog data to templates

### PR DESCRIPTION
**Pull request by sowasred2012**
## Done

- added blog module
- created a tag on insights.ubuntu.com for "Kubeflow News" (id: 3408)
- setup basic templates for:
  - index page (to feature only the latest article)
  - individual articles
  - archives

## QA

- Check out this branch
- Run `./run clean && ./run`
- In your text editor, on line 14 of webapp/app.py, replace:
  - `[3408]` with `[2389]`
  - `'kubeflow-news'` with `'ai'` (because no posts yet have the kubeflow news tag)
- Visit [http://0.0.0.0:8035](http://0.0.0.0:8035), and see that there is a heading and content from the most recent article tagged with 'AI'
- Visit [/archives](http://0.0.0.0:8035/feed), and see that there is a list of articles tagged with 'AI'
- Visit [/feed](http://0.0.0.0:8035/feed), and see that there is an XML list of articles

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+<html class="no-js{% block extra_html_class %}{% endblock %}" lang="en" dir="ltr">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+
+    <title>{% block title %}{{ title }}{% endblock %} | Kubeflow News</title>
+
+    <script>
+      var html = document.documentElement
+      html.classList.replace("no-js", "js");
```
Comment (tbille) [inline templates/base.html]: why is that?

Comment (sowasred2012) [reply templates/base.html]: I carried it over as boilerplate from another project - I thought we'd need it eventually, but I guess we can introduce it later if we do need it. I'll remove.

File: webapp/app.py [hunk 2]
```diff
@@ -11,8 +10,11 @@
     static_folder="../static",
 )
 
+blog = BlogExtension(
+    app, "kubeflow-news.com", [3408], "kubeflow-news", "/", [3184, 3265]
+)
+
 template_finder_view = TemplateFinder.as_view("template_finder")
-app.add_url_rule("/", view_func=template_finder_view)
 app.add_url_rule("/<path:subpath>", view_func=template_finder_view)
```
Comment (tbille) [inline webapp/app.py]: is the template finder still needed?

File: webapp/app.py [hunk 3]
```diff
@@ -1,8 +1,7 @@
 import flask
 from canonicalwebteam.flask_base.app import FlaskBase
-
 from canonicalwebteam.templatefinder import TemplateFinder
-
+from canonicalwebteam.blog.app import BlogExtension
```
Comment (tbille) [inline webapp/app.py]: can you put it in alphabetical order?

File: webapp/app.py [hunk 4]
```diff
@@ -12,6 +12,17 @@
...
+
+
+@app.errorhandler(404)
+def not_found_error(error):
+    return flask.render_template("404.html"), 404
+
+
+@app.errorhandler(500)
+def internal_error(error):
+    return flask.render_template("500.html"), 500
```
Comment (tbille) [inline webapp/app.py]: the error handlers are not needed. the FlaskBase app handles them

File: webapp/app.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
-# Packages
+import flask
```
Comment (tbille) [inline webapp/app.py]: linter

**Pull request merged**